liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -112,11 +117,11 @@ plugin = GHC.defaultPlugin {
     -- See also: https://github.com/ucsd-progsys/liquidhaskell/issues/1727
     -- for a post-mortem.
     liquidPluginGo cfg summary gblEnv = do
+      breadcrumb <- swapBreadcrumb (toStableModule thisModule) (Just Typechecking)
```

Comment:
Should `swapBreadcrumb` be called again when `liquidPluginGo` returns to remove the breadcrumb? For instance,
```
bracket
 (swapBreadcrumb (toStableModule thisModule) (Just Typechecking))
 (const $ swapBreadcrumb (toStableModule thisModule) Nothing)
 $ \breadcrumb ->
```

Reply:
This would mean that the recursive typechecker hook would be the one removing the breadcrumb. I don't like that, because that means we now have this prerequisite that "there is only one level of recursive typechecking" encoded diffusely in the interaction between here and `typecheckedHook`.

Reply:
How is the breadcrumb removed otherwise?

While the breadcrumb is there the plugin cannot verify the module a second time, and I'd find this surprising. This is fine when GHC is running per module or package, but I've heard of experiments to make it run as a compilation server.

Reply:
Oof, I wasn't thinking of running long enough to typecheck the same module multiple times.

In that case, I think the prudent thing to do is to remove the breadcrumb at the end of `typecheckHook`.